paasta_tools/cli/cmds/spark_run.py

```diff
@@ -807,10 +857,27 @@ def paasta_spark_run(args):
     if docker_image is None:
         return 1
 
+    pod_template_path = generate_pod_template_path()
+
+    if args.enable_compact_bin_packing and (
+        not os.access(POD_TEMPLATE_DIR, os.R_OK)
+        or args.cluster_manager != CLUSTER_MANAGER_K8S
+    ):
+        args.enable_compact_bin_packing = False
```

Comment:
Could you please log or throw a warning in case `enable_compact_bin_packing` is enabled but the other checks are not passing? Especially if `os.access` is False-y.

Reply:
Thanks 👍 I've separated this into it's own method `should_enable_compact_bin_packing` and added tests for the different cases.